<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Inference in practice | Phylogenetic Biology</title>
  <meta name="description" content="An introduction to Phylogenetic Biology." />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Inference in practice | Phylogenetic Biology" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="An introduction to Phylogenetic Biology." />
  <meta name="github-repo" content="caseywdunn/phylogenetic_biology" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Inference in practice | Phylogenetic Biology" />
  <meta name="twitter:site" content="@caseywdunn" />
  <meta name="twitter:description" content="An introduction to Phylogenetic Biology." />
  

<meta name="author" content="Casey W. Dunn" />


<meta name="date" content="2020-09-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="inferring-phylogenies-from-data.html"/>
<link rel="next" href="evaluation.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Phylogenetic Biology</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#source-code"><i class="fa fa-check"></i><b>0.1</b> Source code</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#other-resources"><i class="fa fa-check"></i><b>0.2</b> Other resources</a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#software-versions"><i class="fa fa-check"></i><b>0.3</b> Software versions</a></li>
<li class="chapter" data-level="0.4" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>0.4</b> License</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="intro.html"><a href="intro.html#history-of-the-field"><i class="fa fa-check"></i><b>1.1</b> History of the field</a></li>
<li class="chapter" data-level="1.2" data-path="intro.html"><a href="intro.html#core-concepts"><i class="fa fa-check"></i><b>1.2</b> Core concepts</a></li>
<li class="chapter" data-level="1.3" data-path="intro.html"><a href="intro.html#a-unified-perspective-on-phylogenetic-studies"><i class="fa fa-check"></i><b>1.3</b> A unified perspective on phylogenetic studies</a></li>
<li class="chapter" data-level="1.4" data-path="intro.html"><a href="intro.html#applications"><i class="fa fa-check"></i><b>1.4</b> Applications</a></li>
<li class="chapter" data-level="1.5" data-path="intro.html"><a href="intro.html#additional-resources"><i class="fa fa-check"></i><b>1.5</b> Additional resources</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="phylogenies.html"><a href="phylogenies.html"><i class="fa fa-check"></i><b>2</b> Phylogenies</a><ul>
<li class="chapter" data-level="2.1" data-path="phylogenies.html"><a href="phylogenies.html#phylogenies-are-graphs"><i class="fa fa-check"></i><b>2.1</b> Phylogenies are graphs</a></li>
<li class="chapter" data-level="2.2" data-path="phylogenies.html"><a href="phylogenies.html#drawing-phylogenies"><i class="fa fa-check"></i><b>2.2</b> Drawing phylogenies</a></li>
<li class="chapter" data-level="2.3" data-path="phylogenies.html"><a href="phylogenies.html#some-properties-of-trees"><i class="fa fa-check"></i><b>2.3</b> Some properties of trees</a></li>
<li class="chapter" data-level="2.4" data-path="phylogenies.html"><a href="phylogenies.html#rotating-nodes"><i class="fa fa-check"></i><b>2.4</b> Rotating nodes</a></li>
<li class="chapter" data-level="2.5" data-path="phylogenies.html"><a href="phylogenies.html#the-meaning-of-edge-lengths"><i class="fa fa-check"></i><b>2.5</b> The meaning of edge lengths</a></li>
<li class="chapter" data-level="2.6" data-path="phylogenies.html"><a href="phylogenies.html#naming-nodes-and-groups"><i class="fa fa-check"></i><b>2.6</b> Naming nodes and groups</a></li>
<li class="chapter" data-level="2.7" data-path="phylogenies.html"><a href="phylogenies.html#polytomies"><i class="fa fa-check"></i><b>2.7</b> Polytomies</a></li>
<li class="chapter" data-level="2.8" data-path="phylogenies.html"><a href="phylogenies.html#rooting"><i class="fa fa-check"></i><b>2.8</b> Rooting</a></li>
<li class="chapter" data-level="2.9" data-path="phylogenies.html"><a href="phylogenies.html#the-information-contained-in-phylogenies"><i class="fa fa-check"></i><b>2.9</b> The information contained in phylogenies</a></li>
<li class="chapter" data-level="2.10" data-path="phylogenies.html"><a href="phylogenies.html#representation"><i class="fa fa-check"></i><b>2.10</b> Representation</a></li>
<li class="chapter" data-level="2.11" data-path="phylogenies.html"><a href="phylogenies.html#additional-reading"><i class="fa fa-check"></i><b>2.11</b> Additional reading</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="simulation.html"><a href="simulation.html"><i class="fa fa-check"></i><b>3</b> Simulation</a><ul>
<li class="chapter" data-level="3.1" data-path="simulation.html"><a href="simulation.html#models"><i class="fa fa-check"></i><b>3.1</b> Models</a></li>
<li class="chapter" data-level="3.2" data-path="simulation.html"><a href="simulation.html#a-simple-model-of-evolution"><i class="fa fa-check"></i><b>3.2</b> A simple model of evolution</a><ul>
<li class="chapter" data-level="3.2.1" data-path="simulation.html"><a href="simulation.html#expected-amount-of-change"><i class="fa fa-check"></i><b>3.2.1</b> Expected amount of change</a></li>
<li class="chapter" data-level="3.2.2" data-path="simulation.html"><a href="simulation.html#expected-end-state"><i class="fa fa-check"></i><b>3.2.2</b> Expected end state</a></li>
<li class="chapter" data-level="3.2.3" data-path="simulation.html"><a href="simulation.html#analytical-approach"><i class="fa fa-check"></i><b>3.2.3</b> Analytical approach</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="simulation.html"><a href="simulation.html#generalizing-the-simple-model"><i class="fa fa-check"></i><b>3.3</b> Generalizing the simple model</a></li>
<li class="chapter" data-level="3.4" data-path="simulation.html"><a href="simulation.html#expanding-the-models"><i class="fa fa-check"></i><b>3.4</b> Expanding the models</a></li>
<li class="chapter" data-level="3.5" data-path="simulation.html"><a href="simulation.html#plugging-some-numbers-into-the-expanded-model"><i class="fa fa-check"></i><b>3.5</b> Plugging some numbers into the expanded model</a></li>
<li class="chapter" data-level="3.6" data-path="simulation.html"><a href="simulation.html#simulation-along-a-single-edge"><i class="fa fa-check"></i><b>3.6</b> Simulation along a single edge</a></li>
<li class="chapter" data-level="3.7" data-path="simulation.html"><a href="simulation.html#scaling-from-a-single-edge-to-a-whole-tree"><i class="fa fa-check"></i><b>3.7</b> Scaling from a single edge to a whole tree</a></li>
<li class="chapter" data-level="3.8" data-path="simulation.html"><a href="simulation.html#scaling-from-a-single-site-to-multiple-sites"><i class="fa fa-check"></i><b>3.8</b> Scaling from a single site to multiple sites</a></li>
<li class="chapter" data-level="3.9" data-path="simulation.html"><a href="simulation.html#concluding-thoughts"><i class="fa fa-check"></i><b>3.9</b> Concluding thoughts</a></li>
<li class="chapter" data-level="3.10" data-path="intro.html"><a href="intro.html#additional-resources"><i class="fa fa-check"></i><b>3.10</b> Additional resources</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="inferring-phylogenies-from-data.html"><a href="inferring-phylogenies-from-data.html"><i class="fa fa-check"></i><b>4</b> Inferring phylogenies from data</a><ul>
<li class="chapter" data-level="4.1" data-path="inferring-phylogenies-from-data.html"><a href="inferring-phylogenies-from-data.html#probability-of-a-single-history"><i class="fa fa-check"></i><b>4.1</b> Probability of a single history</a></li>
<li class="chapter" data-level="4.2" data-path="inferring-phylogenies-from-data.html"><a href="inferring-phylogenies-from-data.html#probability-of-multiple-histories"><i class="fa fa-check"></i><b>4.2</b> Probability of multiple histories</a></li>
<li class="chapter" data-level="4.3" data-path="inferring-phylogenies-from-data.html"><a href="inferring-phylogenies-from-data.html#log-likelihood"><i class="fa fa-check"></i><b>4.3</b> Log likelihood</a></li>
<li class="chapter" data-level="4.4" data-path="inferring-phylogenies-from-data.html"><a href="inferring-phylogenies-from-data.html#likelihood-for-multiple-sites"><i class="fa fa-check"></i><b>4.4</b> Likelihood for multiple sites</a></li>
<li class="chapter" data-level="4.5" data-path="inferring-phylogenies-from-data.html"><a href="inferring-phylogenies-from-data.html#maximum-likelihood"><i class="fa fa-check"></i><b>4.5</b> Maximum likelihood</a></li>
<li class="chapter" data-level="4.6" data-path="inferring-phylogenies-from-data.html"><a href="inferring-phylogenies-from-data.html#optimality-criteria"><i class="fa fa-check"></i><b>4.6</b> Optimality criteria</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="inference-in-practice.html"><a href="inference-in-practice.html"><i class="fa fa-check"></i><b>5</b> Inference in practice</a><ul>
<li class="chapter" data-level="5.1" data-path="inference-in-practice.html"><a href="inference-in-practice.html#your-question"><i class="fa fa-check"></i><b>5.1</b> Your question</a></li>
<li class="chapter" data-level="5.2" data-path="inference-in-practice.html"><a href="inference-in-practice.html#taxon-sampling"><i class="fa fa-check"></i><b>5.2</b> Taxon sampling</a></li>
<li class="chapter" data-level="5.3" data-path="inference-in-practice.html"><a href="inference-in-practice.html#collecting-sequence-data"><i class="fa fa-check"></i><b>5.3</b> Collecting sequence data</a><ul>
<li class="chapter" data-level="5.3.1" data-path="inference-in-practice.html"><a href="inference-in-practice.html#full-genomes"><i class="fa fa-check"></i><b>5.3.1</b> Full genomes</a></li>
<li class="chapter" data-level="5.3.2" data-path="inference-in-practice.html"><a href="inference-in-practice.html#transcriptomes"><i class="fa fa-check"></i><b>5.3.2</b> Transcriptomes</a></li>
<li class="chapter" data-level="5.3.3" data-path="inference-in-practice.html"><a href="inference-in-practice.html#targetted-enrichment"><i class="fa fa-check"></i><b>5.3.3</b> Targetted enrichment</a></li>
<li class="chapter" data-level="5.3.4" data-path="inference-in-practice.html"><a href="inference-in-practice.html#rad-seq"><i class="fa fa-check"></i><b>5.3.4</b> RAD-seq</a></li>
<li class="chapter" data-level="5.3.5" data-path="inference-in-practice.html"><a href="inference-in-practice.html#pcr"><i class="fa fa-check"></i><b>5.3.5</b> PCR</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="inference-in-practice.html"><a href="inference-in-practice.html#data-processing-upstream-of-phylogenetic-analyses"><i class="fa fa-check"></i><b>5.4</b> Data processing upstream of phylogenetic analyses</a><ul>
<li class="chapter" data-level="5.4.1" data-path="inference-in-practice.html"><a href="inference-in-practice.html#estimation-of-original-sequences"><i class="fa fa-check"></i><b>5.4.1</b> Estimation of original sequences</a></li>
<li class="chapter" data-level="5.4.2" data-path="inference-in-practice.html"><a href="inference-in-practice.html#identification-of-homologous-genome-regions"><i class="fa fa-check"></i><b>5.4.2</b> Identification of homologous genome regions</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="inference-in-practice.html"><a href="inference-in-practice.html#working-with-publicly-available-data"><i class="fa fa-check"></i><b>5.5</b> Working with publicly available data</a></li>
<li class="chapter" data-level="5.6" data-path="inference-in-practice.html"><a href="inference-in-practice.html#homologous-sites"><i class="fa fa-check"></i><b>5.6</b> Homologous sites</a></li>
<li class="chapter" data-level="5.7" data-path="inference-in-practice.html"><a href="inference-in-practice.html#combining-data-across-multiple-genome-regions"><i class="fa fa-check"></i><b>5.7</b> Combining data across multiple genome regions</a></li>
<li class="chapter" data-level="5.8" data-path="inference-in-practice.html"><a href="inference-in-practice.html#model-selection"><i class="fa fa-check"></i><b>5.8</b> Model selection</a></li>
<li class="chapter" data-level="5.9" data-path="inference-in-practice.html"><a href="inference-in-practice.html#running-the-inference"><i class="fa fa-check"></i><b>5.9</b> Running the inference</a></li>
<li class="chapter" data-level="5.10" data-path="inference-in-practice.html"><a href="inference-in-practice.html#viewing-results"><i class="fa fa-check"></i><b>5.10</b> Viewing results</a></li>
<li class="chapter" data-level="5.11" data-path="inference-in-practice.html"><a href="inference-in-practice.html#interpretting-results"><i class="fa fa-check"></i><b>5.11</b> Interpretting results</a></li>
<li class="chapter" data-level="5.12" data-path="inference-in-practice.html"><a href="inference-in-practice.html#computational-considerations"><i class="fa fa-check"></i><b>5.12</b> Computational considerations</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="evaluation.html"><a href="evaluation.html"><i class="fa fa-check"></i><b>6</b> Evaluation</a><ul>
<li class="chapter" data-level="6.1" data-path="evaluation.html"><a href="evaluation.html#model-evaluation"><i class="fa fa-check"></i><b>6.1</b> Model evaluation</a></li>
<li class="chapter" data-level="6.2" data-path="evaluation.html"><a href="evaluation.html#topology-evaluation"><i class="fa fa-check"></i><b>6.2</b> Topology evaluation</a></li>
<li class="chapter" data-level="6.3" data-path="evaluation.html"><a href="evaluation.html#sensitiviy"><i class="fa fa-check"></i><b>6.3</b> Sensitiviy</a></li>
<li class="chapter" data-level="6.4" data-path="evaluation.html"><a href="evaluation.html#confidence"><i class="fa fa-check"></i><b>6.4</b> Confidence</a></li>
<li class="chapter" data-level="6.5" data-path="evaluation.html"><a href="evaluation.html#epistemology"><i class="fa fa-check"></i><b>6.5</b> Epistemology</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="future.html"><a href="future.html"><i class="fa fa-check"></i><b>7</b> Future</a></li>
<li class="chapter" data-level="8" data-path="shape.html"><a href="shape.html"><i class="fa fa-check"></i><b>8</b> Shape</a></li>
<li class="chapter" data-level="9" data-path="character-evolution.html"><a href="character-evolution.html"><i class="fa fa-check"></i><b>9</b> Character evolution</a></li>
<li class="chapter" data-level="10" data-path="intro.html"><a href="intro.html#applications"><i class="fa fa-check"></i><b>10</b> Applications</a><ul>
<li class="chapter" data-level="10.1" data-path="applications.html"><a href="applications.html"><i class="fa fa-check"></i><b>10.1</b> Functional genomic data</a></li>
<li class="chapter" data-level="10.2" data-path="applications.html"><a href="applications.html#gene-trees"><i class="fa fa-check"></i><b>10.2</b> Gene trees</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Phylogenetic Biology</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="inference-in-practice" class="section level1">
<h1><span class="header-section-number">Chapter 5</span> Inference in practice</h1>
<p>In the last chapter we examined the mathematical and statistical underpinning of phylogenetic inference with maximum likelihood. In this chapter we examine the practice of phylogenetic inference – what are the actual steps you need to take to infer topology and edge lengths? We are sticking with DNA sequence data for now, but we will extend these approaches to other types of data later.</p>
<p>This is not a step-by-step guide to how to click through an analysis. It is an examination of the various issues and decisions you will encounter in the design and execution of a typical analysis.</p>
<div id="your-question" class="section level2">
<h2><span class="header-section-number">5.1</span> Your question</h2>
<p>The first step to an analysis is to clearly define your question. There will be many decisions to make along the way, and a clear question will provide an explicit benchmark for working through these decisions – which decision is best for the question at hand? If you have a vague question it will be much harder to decide which steps are best, and many decisions about tradeoffs will essentially be arbitrary.</p>
</div>
<div id="taxon-sampling" class="section level2">
<h2><span class="header-section-number">5.2</span> Taxon sampling</h2>
<p>Phylogenetic questions are almost always taxon specific, so the first thing you will want to consider is taxon sampling.</p>
<p>One of the most common mistakes in phylogenetic analyses is to include too few taxa (ie, tips). There are many ways that adding taxa helps:</p>
<ul>
<li><p>Adding taxa breaks up edges in the phylogeny. This often improves the performance of maximum likelihood. So if you are only interested in taxa 1…20, it still helps to add in taxa 21…50, even if you end up not paying much attention to where they go. They will often improve your ability to infer relationships among the taxa you are interested in.</p></li>
<li><p>Adding taxa can help improve model evaluation and model parameter estimation. Like breaking up edges, this is a mechanism by which adding taxa that you don’t care so much about can improve the analysis of the taxa you do care about.</p></li>
<li><p>Adding taxa reduces the number of assumptions going into a study. You are almost always inferring a phylogeny because you are uncertain at the outset about evolutionary relationships. If, for example, you only include a couple of species from Clade B because you assume it is monophyletic, this sampling may be insufficient to reveal that Clade C is in fact nested within Clade B. With insufficient sampling of Clade B, it may instead appear that the two groups are sister clades.</p></li>
<li><p>Adding taxa provides a more rigorous test of phylogenetic hypotheses. Say you are conducting a study to investigate the phylogenetic placement of a specific enigmatic species that is so morphologically distinct from other organisms that it is hard to know where it goes. You think it may fall in Clade A, Clade B, or Clade C, so you densely sample those groups. But in fact it falls within Clade D, and our taxon sampling is such that you can’t even evaluate this hypothesis because you have no or too few taxa from Clade D in your study.</p></li>
<li><p>Adding more outgroup taxa often improves rooting. Skimping on outgroup sampling is one of the most common mistakes in project design. This can lead to rooting problems. The best practice in general is to include the most closely related outgroup taxa possible, and some that are progressively more distantly related. The root and the edges that connect to it are the deepest in the tree, and can be the hardest to infer. So it is good to densely sample taxa relevant to the root.</p></li>
</ul>
<p>There are costs to adding taxa that must be balanced against the benefits above:</p>
<ul>
<li><p>If you are collecting the data, it costs money to add taxa. It can be very expensive to collect the specimens and then to sequence them, depending on the approach you are taking. That money may be best allocated in other ways.</p></li>
<li><p>Adding taxa can considerably increase the computational demand of analyses. This can increase the time and cost of analysis, and maybe even make an analysis unfeasible on the computers available to the investigator.</p></li>
</ul>
<p>There are some conditions under which you may want to exclude specific taxa, at least from some analyses:</p>
<ul>
<li><p>If you suspect that they data are contaminated, incorrectly identified, or insufficiently documented.</p></li>
<li><p>Preliminary analyses suggest that the taxon has properties very different from other taxa in the study, such as a very different equilibrium frequency or an exceptionally high rate of evolution.</p></li>
<li><p>The taxon has a very high fraction of missing data. Sometimes missing data makes a taxon unstable, which can obscure well supported relationships between other taxa. Sometimes missing data can interact with other factors to lead to bias in the placement of the taxon.</p></li>
</ul>
</div>
<div id="collecting-sequence-data" class="section level2">
<h2><span class="header-section-number">5.3</span> Collecting sequence data</h2>
<p>Once you have a good idea about taxon sampling, you will need to consider which genome regions to include in your analyses. To build a phylogeny, you need homologous characters across the organism under investigation. In the context of DNA analyses, this means that you need overlapping genome regions across the taxa in your tree.</p>
<p>Decisions about which taxa include interact quite a bit with decisions about which genome regions you will consider. So there is often quite a bit of back and forth as you refine these decisions. At the coarsest level, if you are collecting data yourself you will often need to decide if you are going to capture a larger amount of data from fewer taxa or s smaller amount of data from many taxa. If you are using publicly available data, then your taxon and genome region sampling will be strongly constrained by what data are already available.</p>
<p>Decisions about what genome regions to include can be thought of as decisions about genome sequence enrichment strategies that can restrict data acquisition to particular regions. Here I describe common enrichment methods along a spectrum of increasing enrichment.</p>
<div id="full-genomes" class="section level3">
<h3><span class="header-section-number">5.3.1</span> Full genomes</h3>
<p>Phylogenies are now routinely inferred without enrichment of any genome regions - the full genome is sequenced and analyzed. This is possible in large part to increasing read length, higher quality, and declining cost of new DNA sequencing technologies. Full genome phylogenetic analyses are still largely restricted to organisms with small genomes, in particular viruses. This has been a very powerful tool, for example, to investigate virus transmission through human populations. It is still expensive and difficult to build phylogenies with large genome sequences, though this is rapidly changing.</p>
</div>
<div id="transcriptomes" class="section level3">
<h3><span class="header-section-number">5.3.2</span> Transcriptomes</h3>
<p>A transcriptome is the collection of RNA in a cell or collection of cells. Because RNA is transcribed from the genome, sequencing the RNA gives a snapshot of an enriched subset of the genome. Whereas the other enrichment approaches we will see below are done bby the investigator at the bench, transcriptome sequencing largely relies on a natrural process in the living cell (transcription) to enrich for particular genome regions.</p>
<p>In a typical phylogenetic study based on transcriptome data, messenger RNA (mRNA), which encode the protein coding genes, is isolated. mRNA is then copied to complimentary DNA (cDNA) and sequenced. These DNA sequences are then assembled into predictions of protein coding genes. Because many of the same genes are highly expressed across species and tissues, there is considerable overlap between the genes in transcriptome datasets. Many studies are interested in protein coding genes for other purposes as well, so transcriptomes are a very cost-effective way to collect data from a broad diversity of species that can be used for multiple purposes.</p>
<p>There are a couple challenges to working with transcriptomes, though. For one, RNA is much less stable than DNA. It is therefore critical to take considerable care when collecting, storing, and processing tissue for transcriptome studies.</p>
</div>
<div id="targetted-enrichment" class="section level3">
<h3><span class="header-section-number">5.3.3</span> Targetted enrichment</h3>
<p>In targeted enrichment, and investigator designs short bait sequences that are similar to conserved genome regions. They then isolate DNA from their taxa of interest, fragment it, and use the baits to fish out the conserved regions along with their neighboring sequences. This allows the investigator to created pools of DNA that are enriched for specific pre-selected regions. This is a highly cost effective way to isolate homologous sequences across species, and works well when samples have fragmented DNA (in museum specimens, for example).</p>
<p>The bates are typically designed based on one or more genomes that are already available in the clade of interest. There are a few challenges using targeted enrichment. One of the most challenging is that data are difficult to combine across different studies that used different baits. In addition, the baits can be challenging to design and costly to synthesize. These up-front investments make this approach well suited to larger projects.</p>
</div>
<div id="rad-seq" class="section level3">
<h3><span class="header-section-number">5.3.4</span> RAD-seq</h3>
<p>Like targeted enrichment, RAD-seq enriches for sequence data collected across the genome. It differs, though, in that it uses intrinsic properties of the genome for enrichment, rather than user-designed baits. Genomic DNA is digested with one or two restriction enzymes that cut the genome at specific small sequences, and leave the DNA with particular overhangs at the cut end. These cut fragments are then size selected. If cutting with two restriction enzymes, the fragments that have different overhangs (because they were cut once with each enzyme) are then further enriched and sequenced.</p>
<p>The result is that sequences are enriched for the same regions across taxa, as long as the same restriction enzyme sites occur with the same spacing in the genome. This is a pwoerful way to get a consistent subset of the genome across many different samples. It doesn’t require a big investment in baits at the outset, but there are a few important limitations. First, it only works well on taxa that are quite closely related, since differences in restriction site placement evolve relatively quickly and enrichment overlap falls quickly with evolutionary distance. Rad-seq therefore is most often used when looking at evolutionary relationships within species or among very closely related species. Second, there is very little control over which specific genome regions are sequenced. For example, if you want to build a phylogeny and look at the evolution of a particular gene, you can’t readily engineer the sample enrichment to enrich for that specific gene.</p>
</div>
<div id="pcr" class="section level3">
<h3><span class="header-section-number">5.3.5</span> PCR</h3>
<p>The most extreme form of genome enrichment is Polymerase Chain Reaction (PCR). This method enriches for a small number of specific genome regions selected by the user. Short DNA oligonucleotide sequences are designed to flank the region of interest and target it for amplification. By placing these oligonucleotides in conserved regions and engineering them to bind slightly different sequences, they can be used to successfully enrich the same genome region from quite distantly related species. PCR enabled the first wave of DNA sequence-based phylogenetic inference starting in the 1980s, and is still widely used used today even has a variety of new enrichment methods have become available.</p>
<p>PCR is cheap and relatively easy, and because it has been used for decades to sequence a small number of genes across a wide diversity of organisms, it is a great way to collect data that can be readily analyzed in combination with publicly available data. Ironically, though, it is now very expensive per-nucleotide to collect data with PCR relative to the other methods above. That’s because it is difficult to leverage the advantages of modern high-throughput sequencing, which is very cheap per nucleotide, with such a small number of short amplified sequences. As one of the most extreme forms of genome enrichment available, it provides access to a very small fraction of the genome. PCR is a convenient way to get a tiny amount of sequence data from a large number of taxa.</p>
</div>
</div>
<div id="data-processing-upstream-of-phylogenetic-analyses" class="section level2">
<h2><span class="header-section-number">5.4</span> Data processing upstream of phylogenetic analyses</h2>
<p>The above section describes how different enrichment strategies work when collecting sequence data for phylogenetic analyses, but the differences are not limited to what happens at the bench. Each enrichment strategy requires its own series of data processing steps to get the data ready for phylogenetic inference. These steps vary greatly across enrichment approaches, requiring distinct methods and software tools. Exploring them in detail is beyond the scope of this text, but there are a few general aspects I will address.</p>
<p>The overall goal of the data processing that occurs upstream of the actual phylogenetic analysis has two primary objectives, estimation of original genome sequences and identification of homologous sequences across species.</p>
<div id="estimation-of-original-sequences" class="section level3">
<h3><span class="header-section-number">5.4.1</span> Estimation of original sequences</h3>
<p>Raw sequence data have a few properties that make them unusable for direct analysis:</p>
<ul>
<li><p>They are redundant. The same genome regions are sequenced multiple times. PCR products are typically sequenced a couple times (once in each direction), and high throughput sequencing often sequences each region hundreds or even thousands of times.</p></li>
<li><p>They have many errors. These are introduced during sample preparation at the bench, by the sequencing instruments during data acquisition, and in the initial base calling that generates the sequence file.</p></li>
<li><p>They are fragmented. Many times toe sequence reads do not span the full genome region of interest, and must be tiled together to predict longer regions.</p></li>
<li><p>They are ambiguous. It often isn’t clear what region of the genome a sequence belongs to, due to genomic repeats, heterozygosity, and genome duplication. This ambiguity must be resolved or eliminated, depending on the application.</p></li>
</ul>
<p>The above technical challenges are usually addressed simultaneously in an analysis process called assembly. The goal of assembly is to predict the sequences of the original genome regions based on the raw sequence data. The general idea is to line up overlapping raw sequence data, correct errors by comparing sequences, and then to collapse all the sequences down to hypotheses about the original actual sequence from which all the raw data were generated. This is a very complex process, and still a major challenge for many genomes and transcriptomes in particular. The best approach differs quite a bit depending on the enrichment strategy, specific type of data collected, and the properties of the genome.</p>
<p>The process of assembling a genome (or genome regions) from scratch directly from the raw sequence data is referred to as <em>de novo</em> assembly. If high quality closely related genomes are available, they can be used for guided assembly or mapping, where the raw data are tiled onto the reference to rather than assembled from scratch.</p>
</div>
<div id="identification-of-homologous-genome-regions" class="section level3">
<h3><span class="header-section-number">5.4.2</span> Identification of homologous genome regions</h3>
<p>It isn’t enough to collect and assemble sequence data from our taxa, we need to collect and identify <em>homologous</em> data that represent regions that are shared because they derive the the same region in common ancestors of the taxa.</p>
<p>Like assembly, the optimal approach for identifying homologous sequences depends greatly on the enrichment strategy used. This step is quite straight forward when targetted capture and PCR are used as enrichment strategies, because the data are from specific know regions that were often selected in part because their homology is easier to assess.</p>
<p>In some cases, such as the closely related virus genomes, genomes can be directly aligned with each other along their full length. In larger genomes, and transcriptomes, protein coding sequences are often identified and used in further analyses. There are various approaches to identifying homologous protein coding genes across taxa. One is to identify sequences for a pre-selected panel of genes that are selected because they tend to be present in most datasets, they have properties of molecular evolution that are well-suited to phylogenetic analysis, and their homology is thought to be easier to assess. Another is to compare all gene sequences across all species to identify homologous regions. This can be a bit more computational work, but has the advantage of being able to optimize gene selection to the data and question at hand.</p>
</div>
</div>
<div id="working-with-publicly-available-data" class="section level2">
<h2><span class="header-section-number">5.5</span> Working with publicly available data</h2>
<p>Most phylogenetic studies include at least some publicly available data from previously published studies. Some rely entirely on publicly available data, while others add new sequence data as well. The primary repositories for DNA sequence data in published studies, including phylogenetic studies, are <a href="https://www.ncbi.nlm.nih.gov/">NCBI</a> and <a href="https://www.ebi.ac.uk/">EMBL-EBI</a>. Many of the same data are available in each.</p>
<p>Most phylogenetic studies deposit their raw data in public archives like the <a href="https://www.ncbi.nlm.nih.gov/sra">NCBI Sequence Read Archive</a>, but these can require considerable processing (including assembly) to include in a phylogenetic inference. It can be hard to track down assemblies and other intermediate products that are easier to work with since they are not consistently deposited in centralized data archives. Wrangling publicly available data and getting it ready for analysis can be one of the most time consuming components of a phylogenetic inference project. This is one of the major challenges to conducting phylogenetic analyses of DNA sequence data at this time.</p>
<p>When I am interested in building a phylogeny for a particular group of organisms, one of my first steps is to take a look at the <a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi">NCBI taxonomy browser</a>. There you can enter a taxonomic group name, and select which data you would like to see (Figure <a href="inference-in-practice.html#fig:practice-cnidaria">5.1</a>). You can adjust the number of taxonomic levels to control whether you want to view just summaries for larger groups or get all the way down to the species level.</p>
<div class="figure"><span id="fig:practice-cnidaria"></span>
<img src="figures/cnidaria_taxonomy_browser.png" alt="A view of the sequence data available for Cnidaria. Here I have selected to view available Nucleotide, Protein, SRA, Genome, and Assembly data."  />
<p class="caption">
Figure 5.1: A view of the sequence data available for Cnidaria. Here I have selected to view available Nucleotide, Protein, SRA, Genome, and Assembly data.
</p>
</div>
<p>Based on this overview, you can assess which types of data (<em>i.e.</em>, which data produced by which enrichment strategy) are best for advancing your question with publicly available data. All the PCR data and most of the raw sequence data will usually be available here, but many assemblies and processed data for whole genomes, transcriptomes, and other high-throughput datasets may not be. The best practice in these cases is to take a deep dive into the recent literature for the group and see if these intermediate products were deposited in another public archive, or contact authors for more details.</p>
</div>
<div id="homologous-sites" class="section level2">
<h2><span class="header-section-number">5.6</span> Homologous sites</h2>
<p>Once you have collected homologous sequences for one or more genome sequences across your taxa of interest, you next need to identify homologous nucleotide sites within these regions. This process is referred to as multiple sequence alignment (MSA). Rather than rey on explicit models of molecular evolution, multiple sequence alignment usually relies on similarity of sequences. The general idea is to slide the sequences along each other and insert gaps as needed so that similarity at each site is optimized while reducing the number of gaps introduced.</p>
<p>The output of multiple sequence alignment is a matrix, where rows are taxa that will correspond to tips in the phylogeny and each column is a single nucleotide site that is hypothesized to be homologous across taxa.</p>
</div>
<div id="combining-data-across-multiple-genome-regions" class="section level2">
<h2><span class="header-section-number">5.7</span> Combining data across multiple genome regions</h2>
</div>
<div id="model-selection" class="section level2">
<h2><span class="header-section-number">5.8</span> Model selection</h2>
</div>
<div id="running-the-inference" class="section level2">
<h2><span class="header-section-number">5.9</span> Running the inference</h2>
</div>
<div id="viewing-results" class="section level2">
<h2><span class="header-section-number">5.10</span> Viewing results</h2>
</div>
<div id="interpretting-results" class="section level2">
<h2><span class="header-section-number">5.11</span> Interpretting results</h2>
</div>
<div id="computational-considerations" class="section level2">
<h2><span class="header-section-number">5.12</span> Computational considerations</h2>



</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="inferring-phylogenies-from-data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="evaluation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["phylogenetic_biology.pdf", "phylogenetic_biology.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
