<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 Introduction | Phylogenetic Biology</title>
  <meta name="description" content="An introduction to Phylogenetic Biology." />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 Introduction | Phylogenetic Biology" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="An introduction to Phylogenetic Biology." />
  <meta name="github-repo" content="caseywdunn/phylogenetic_biology" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 Introduction | Phylogenetic Biology" />
  <meta name="twitter:site" content="@caseywdunn" />
  <meta name="twitter:description" content="An introduction to Phylogenetic Biology." />
  

<meta name="author" content="Casey W. Dunn" />


<meta name="date" content="2020-08-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="phylogenies.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#source-code"><i class="fa fa-check"></i><b>0.1</b> Source code</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#software-versions"><i class="fa fa-check"></i><b>0.2</b> Software versions</a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>0.3</b> License</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="intro.html"><a href="intro.html#history-of-the-field"><i class="fa fa-check"></i><b>1.1</b> History of the field</a></li>
<li class="chapter" data-level="1.2" data-path="intro.html"><a href="intro.html#core-concepts"><i class="fa fa-check"></i><b>1.2</b> Core concepts</a></li>
<li class="chapter" data-level="1.3" data-path="intro.html"><a href="intro.html#a-unified-perspective-on-phylogenetic-studies"><i class="fa fa-check"></i><b>1.3</b> A unified perspective on phylogenetic studies</a></li>
<li class="chapter" data-level="1.4" data-path="intro.html"><a href="intro.html#applications"><i class="fa fa-check"></i><b>1.4</b> Applications</a></li>
<li class="chapter" data-level="1.5" data-path="intro.html"><a href="intro.html#additional-resources"><i class="fa fa-check"></i><b>1.5</b> Additional resources</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="phylogenies.html"><a href="phylogenies.html"><i class="fa fa-check"></i><b>2</b> Phylogenies</a><ul>
<li class="chapter" data-level="2.1" data-path="phylogenies.html"><a href="phylogenies.html#phylogenies-are-graphs"><i class="fa fa-check"></i><b>2.1</b> Phylogenies are graphs</a></li>
<li class="chapter" data-level="2.2" data-path="phylogenies.html"><a href="phylogenies.html#drawing-phylogenies"><i class="fa fa-check"></i><b>2.2</b> Drawing phylogenies</a></li>
<li class="chapter" data-level="2.3" data-path="phylogenies.html"><a href="phylogenies.html#some-properties-of-trees"><i class="fa fa-check"></i><b>2.3</b> Some properties of trees</a></li>
<li class="chapter" data-level="2.4" data-path="phylogenies.html"><a href="phylogenies.html#rotating-nodes"><i class="fa fa-check"></i><b>2.4</b> Rotating nodes</a></li>
<li class="chapter" data-level="2.5" data-path="phylogenies.html"><a href="phylogenies.html#the-meaning-of-edge-lengths"><i class="fa fa-check"></i><b>2.5</b> The meaning of edge lengths</a></li>
<li class="chapter" data-level="2.6" data-path="phylogenies.html"><a href="phylogenies.html#naming-nodes-and-groups"><i class="fa fa-check"></i><b>2.6</b> Naming nodes and groups</a></li>
<li class="chapter" data-level="2.7" data-path="phylogenies.html"><a href="phylogenies.html#polytomies"><i class="fa fa-check"></i><b>2.7</b> Polytomies</a></li>
<li class="chapter" data-level="2.8" data-path="phylogenies.html"><a href="phylogenies.html#rooting"><i class="fa fa-check"></i><b>2.8</b> Rooting</a></li>
<li class="chapter" data-level="2.9" data-path="phylogenies.html"><a href="phylogenies.html#the-information-contained-in-phylogenies"><i class="fa fa-check"></i><b>2.9</b> The information contained in phylogenies</a></li>
<li class="chapter" data-level="2.10" data-path="phylogenies.html"><a href="phylogenies.html#representation"><i class="fa fa-check"></i><b>2.10</b> Representation</a></li>
<li class="chapter" data-level="2.11" data-path="phylogenies.html"><a href="phylogenies.html#additional-reading"><i class="fa fa-check"></i><b>2.11</b> Additional reading</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="simulation.html"><a href="simulation.html"><i class="fa fa-check"></i><b>3</b> Simulation</a><ul>
<li class="chapter" data-level="3.1" data-path="simulation.html"><a href="simulation.html#models"><i class="fa fa-check"></i><b>3.1</b> Models</a></li>
<li class="chapter" data-level="3.2" data-path="simulation.html"><a href="simulation.html#a-simple-model"><i class="fa fa-check"></i><b>3.2</b> A simple model</a><ul>
<li class="chapter" data-level="3.2.1" data-path="simulation.html"><a href="simulation.html#expected-amount-of-change"><i class="fa fa-check"></i><b>3.2.1</b> Expected amount of change</a></li>
<li class="chapter" data-level="3.2.2" data-path="simulation.html"><a href="simulation.html#expected-end-state"><i class="fa fa-check"></i><b>3.2.2</b> Expected end state</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="simulation.html"><a href="simulation.html#generalizing-the-simple-model"><i class="fa fa-check"></i><b>3.3</b> Generalizing the simple model</a></li>
<li class="chapter" data-level="3.4" data-path="simulation.html"><a href="simulation.html#more-complex-models"><i class="fa fa-check"></i><b>3.4</b> More complex models</a></li>
<li class="chapter" data-level="3.5" data-path="simulation.html"><a href="simulation.html#model-structure"><i class="fa fa-check"></i><b>3.5</b> Model structure</a></li>
<li class="chapter" data-level="3.6" data-path="simulation.html"><a href="simulation.html#aditional-resources"><i class="fa fa-check"></i><b>3.6</b> Aditional resources</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="inference.html"><a href="inference.html"><i class="fa fa-check"></i><b>4</b> Inference</a></li>
<li class="chapter" data-level="5" data-path="evaluation.html"><a href="evaluation.html"><i class="fa fa-check"></i><b>5</b> Evaluation</a><ul>
<li class="chapter" data-level="5.1" data-path="evaluation.html"><a href="evaluation.html#model-evaluation"><i class="fa fa-check"></i><b>5.1</b> Model evaluation</a></li>
<li class="chapter" data-level="5.2" data-path="evaluation.html"><a href="evaluation.html#topology-evaluation"><i class="fa fa-check"></i><b>5.2</b> Topology evaluation</a></li>
<li class="chapter" data-level="5.3" data-path="evaluation.html"><a href="evaluation.html#sensitiviy"><i class="fa fa-check"></i><b>5.3</b> Sensitiviy</a></li>
<li class="chapter" data-level="5.4" data-path="evaluation.html"><a href="evaluation.html#confidence"><i class="fa fa-check"></i><b>5.4</b> Confidence</a></li>
<li class="chapter" data-level="5.5" data-path="evaluation.html"><a href="evaluation.html#epistemology"><i class="fa fa-check"></i><b>5.5</b> Epistemology</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="future.html"><a href="future.html"><i class="fa fa-check"></i><b>6</b> Future</a></li>
<li class="chapter" data-level="7" data-path="shape.html"><a href="shape.html"><i class="fa fa-check"></i><b>7</b> Shape</a></li>
<li class="chapter" data-level="8" data-path="character-evolution.html"><a href="character-evolution.html"><i class="fa fa-check"></i><b>8</b> Character evolution</a></li>
<li class="chapter" data-level="9" data-path="intro.html"><a href="intro.html#applications"><i class="fa fa-check"></i><b>9</b> Applications</a><ul>
<li class="chapter" data-level="9.1" data-path="applications.html"><a href="applications.html"><i class="fa fa-check"></i><b>9.1</b> Functional genomic data</a></li>
<li class="chapter" data-level="9.2" data-path="applications.html"><a href="applications.html#gene-trees"><i class="fa fa-check"></i><b>9.2</b> Gene trees</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Phylogenetic Biology</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="intro" class="section level1">
<h1><span class="header-section-number">Chapter 1</span> Introduction</h1>
<p>Phylogenetic biology is the study of evolutionary relationships, and the use of those relationships to study other aspects of biology.</p>
<div id="history-of-the-field" class="section level2">
<h2><span class="header-section-number">1.1</span> History of the field</h2>
<p>At the dawn of the field, in the second half of the 19th century <span class="citation">(Darwin <a href="#ref-darwin1859" role="doc-biblioref">1859</a>; Haeckel <a href="#ref-haeckel1897" role="doc-biblioref">1897</a>)</span>, phylogenetic biology was largely a speculative endeavor. People looked at similarities and differences between organisms, and made hypotheses about the evolutionary relationships that would give rise to those patterns. The challenge, though, was that there wasn’t an explicit method for testing those hypotheses. So while there was extensive interest in evolutionary relationships, for many decades the general sense was that most relationships were unknowable. Some evolutionary biologists focused on questions that they believed could be pursued without knowing the specifics of relationships, and others did important work collecting data that would help inform our understandings of those relationships.</p>
<p>It wasn’t until the second half of the 20th century that explicit methods were developed for inferring and testing phylogenetic relationships based on data from organisms <span class="citation">(Hennig <a href="#ref-hennig1966" role="doc-biblioref">1966</a>)</span>. This ushered in decades of accelerating phylogenetic methods development that continues to this day. The development of explicit model based approaches to phylogenetic inference was particularly influential <span class="citation">(Felsenstein <a href="#ref-Felsenstein:1981vk" role="doc-biblioref">1981</a>)</span>. These new methods were highly computationally intensive, but fortunately their rise paralleled the rapid development and widespread availability of computers.</p>
<p>Beginning in the late 1980s, Molecular data became widely used for building phylogenies of extant organisms, in large part because the development of PCR and the declining cost of sequencing made it tractable to collect sequence fragments from a few genes across a broad diversity of organisms. Starting in about 2010, new generations of high-throughput sequencing made it possible to collect sequences for thousands of genes from a broad diversity of organisms in a single study, and we are right now in the earliest days of building phylogenies from high-quality chromosome-level genome assemblies.</p>
<p>Phylogenetic Biology has always been concerned with building phylogenies – using character data collected from a diversity of organisms to study their relationships. This domain is referred to as phylogenetic inference. Coming from its roots in comparative morphology, there was always great interest in learning about evolutionary relationships, using these to improve the classification of species, and examining the major changes in morphology through time.</p>
<p>It has also become clear that the value of studying evolutionary relationships goes well beyond learning about evolutionary relationships. Once so many advances were made in building trees, many investigators have of course asked themselves, “Now what do I do with this tree?” There has also been a growing recognition that many questions require knowing evolutionary relationships, even if those evolutionary relationships are themselves not of central interest to the investigator. This has led to the rise of another domain often referred to as phylogenetic comparative biology. Instead of ending with trees, as many phylogenetic inference projects do, phylogenetic comparative biology projects usually start with trees and use them to study the evolution of traits. Questions include things like whether there is evolutionary covariance between traits, or shifts in rates of evolution. As comparative methods have become more powerful, more phylogenies have been published, and more investigators have recognized that there is a phylogenetic component to their questions, phylogenetic comparative methods have become an increasingly large part of the field.</p>
<p>Though phylogeny initially sprung from the field of comparative morphology, and much of the initial focus was on building and analyzing trees with morphological data and using them to study the evolution of morphology, the field has rapidly expanded to encompass many other categories of data. Even so, morphology continues to be vital to building phylogenies in many contexts, including when fossils for for which no sequence data are available. While most phylogenetic inference is now based on molecular data, and sometimes molecular data, phylogenetic comparative methods are applied to all sorts of data and questions. It now plays an important role in the study of physiology, ecology, genomics, medicine, and most other parts of biology.</p>
<p>Phylogenetics is no longer a strictly macroevolutionary field. Some of the most exciting work in recent years has been at the interface of population genetics and phylogenetic biology, helping to unify our perspectives on micro and macroevolution. Phylogenies are now also routinely used to study the evolution of genes and other molecular characters, and also to dive into extremely recent and fast evolutionary processes within species, such as virus outbreaks.</p>
</div>
<div id="core-concepts" class="section level2">
<h2><span class="header-section-number">1.2</span> Core concepts</h2>
<p>There are a few concepts that are fundamental to understanding phylogenetic analyses. Consider the
variables at play, and their mathematical relationships. These typically include:</p>
<ul>
<li><p>The topology of the phylogeny. This is the structure of the evolutionary relationships.</p></li>
<li><p>The lengths of the branches in the phylogeny. These can mean different things, such as time elapsed or amount of expected change in a character.</p></li>
<li><p>Characters. The organism attributes under consideration. These include things like leg length, geographic location, a particular site in a particular gene, a physiological attribute, or a feature of genome organization.</p></li>
<li><p>Character states. The particular values that can be (or are) taken by different individuals for specific characters. A leg length state could be 4.3mm. For a DNA sequence a character state could be <code>C</code>, <code>G</code>, <code>T</code>, or <code>A</code>.</p></li>
<li><p>Models. Models are hypotheses about how characters evolve. They take a mathematical and statistical form. For example, a model could describe the covariance between evolutionary changes in different characters, or be a set of functions that describe the probabilities of specific state changes. Models are abstract representations of processes, deliberate simplifications that allows us to explicitly describe what we think the most important features of change are. We can make models as simple or complex as we like, but the more complex they are the more information we need to use them effectively.</p></li>
<li><p>Model parameter values. These are the specific values for terms in the model. In a model that indicates covariance between characters, the specific values in such a covariance matrix are model parameter values. If you have a set of functions that describe the probability of changing between particular states for a character, the model parameters are the rates and other terms in those equations.</p></li>
</ul>
<p>Models and model parameters together allow us, for example, to assess the probability of a particular character state to a different character state along a branch of a given length.</p>
<p>In order to make informative comparisons across species, both for inferring phylogenies and for phylogenetic comparative analyses, we need a unified ontological framework to refer to the same characters and character states in different organisms. This correspondence is provided by the concept of homology. Homology is a hypothesis that the same attribute is present in different entities because that attribute was present in their shared ancestor. A deceptively simple concept, it can sometimes be devilishly difficult to define, test, and apply [wagner2014].</p>
</div>
<div id="a-unified-perspective-on-phylogenetic-studies" class="section level2">
<h2><span class="header-section-number">1.3</span> A unified perspective on phylogenetic studies</h2>
<p>Phylogenetics is such a diverse and quickly growing field that sometimes different studies can seem to be more different than they are. This is too bad, as it is a lost opportunity to share ideas across domains and to apply what is learned in one context, say phylogenetic inference, to another context, such as phylogenetic comparative analyses.</p>
<p>Fortunately, phylogenetics has a strong conceptual foundation that provides a unified perspective on phylogenetic studies <span class="citation">(Hohna et al. <a href="#ref-Hohna2014" role="doc-biblioref">2014</a>)</span>. Once you have this in mind, you can quickly see the relationships between different methods, questions, and analyses.</p>
<p>Let’s revisit the list above, and rearrange it a bit:</p>
<ul>
<li>Topology</li>
<li>Branch lengths</li>
<li>Characters</li>
<li>Character states of tips</li>
<li>Character states of ancestors</li>
<li>Model</li>
<li>Model parameters</li>
</ul>
<p>The investigator can generally take one of three approaches to each of these things. They either:</p>
<ul>
<li>Fix the value according to information in hand, such as their own data or the results of a previous study.</li>
<li>Estimate the value.</li>
<li>Don’t care.</li>
</ul>
<p>Many phylogenetic inference studies use expert knowledge to define and fix the characters, run some preliminary analyses to evaluate and then fix the model, and fix the character states at the tips according to their observed data. They then estimate the topology, branch lengths, character states of ancestors, and model parameters. They then basically throw away the estimates of ancestral character states and model parameters, and present the topology and branch lengths as the result. You get a tree that provides an educated inference of the evolutionary relationships between your organisms of interest.</p>
<p>Many phylogenetic comparative analyses take the same approach to characters and observed character states at the tip (both fixed), and ancestral character states (estimated, but discarded). They then take a different approach to the remaining items. They fix the tree and branch lengths according to the results of an earlier phylogenetic inference, evaluate different models, and estimate model parameters. The models and model parameters are then the presented results. They would report, for example, that a model that allows for shifts in rates of evolution along different branches is a better fit than a model that doesn’t, or provide estimates for the evolutionary covariance between particular traits of interest.</p>
<p>Other studies apply the three approaches (fix, estimate, don’t care) in different ways across the phylogenetic study. Some combinations are very common, others very rare.</p>
<p>Once you dig in, the “don’t care” case is surprisingly interesting. Sometimes if you don’t care about something, you don’t need to estimate it. But usually you do need to estimate it to get what you do care about, and then you just throw away the result. In many cases, different analyses differ only in which estimates are kept and which are thrown away. One may keep ancestral character states and discard model parameters, and another may do the exact same thing but discard ancestral character states but keep model parameters</p>
</div>
<div id="applications" class="section level2">
<h2><span class="header-section-number">1.4</span> Applications</h2>
<p>With this framework in mind, let’s take a look at a sampling of recent phylogenetic work.</p>
<ul>
<li><p><a href="https://nextstrain.org/sars-cov-2/" class="uri">https://nextstrain.org/sars-cov-2/</a> provides a frequently updated phylogeny of sars-cov-2, with associated data on geography, sampling time, and other factors. This has been critical to understanding and monitoring the sars-cov-2 pandemic, and intervening to disrupt transmission. They are using observed character data at the tips (genome sequences) to estimate topology, branch lengths, and model parameters.</p></li>
<li><p>Myxozoa are enigmatic parasites that live in fish. They have very few morphological traits that show clear homology with other animals. So Sally Chang, Paulyn Cartwright, and colleagues used phylogenetic inference to examine their relationships to other animals, and found strong support for their placement within Cnidaria. <a href="https://doi.org/10.1073/pnas.1511468112" class="uri">https://doi.org/10.1073/pnas.1511468112</a></p></li>
<li><p>My own lab wanted to see if the evolution of gene expression is more rapid after gene duplication than after speciation. We found no evidence for a different rate of evolution after gene duplication. We used existing gene phylogenies, existing expression data as character states at the tips, and estimated ancestral expression states to examine sifts in expression along particular branches. <a href="https://doi.org/10.1073/pnas.1707515115" class="uri">https://doi.org/10.1073/pnas.1707515115</a></p></li>
<li><p><em>Viburnum</em> is a group of plants with wide distribution across the globe, in many different habitat types. Michael Landis spearheaded a collaborative project between the Edwards and Donoghue labs here at Yale EEB to understand how <em>Viburnum</em> first diversified in Asia during the Eocene and then spread across the globe. They developed innovative methods to simultaneously estimate the phylogeny based on genetic data available at the tips, estimate time along branch lengths with fossils calibrations, and reconstruct geographic range. This is the vanguard of new approaches that simultaneously address multiple categories of data rather than daisy chain together multiple methods. <a href="https://doi.org/10.1093/sysbio/syaa027" class="uri">https://doi.org/10.1093/sysbio/syaa027</a></p></li>
</ul>
</div>
<div id="additional-resources" class="section level2">
<h2><span class="header-section-number">1.5</span> Additional resources</h2>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-darwin1859">
<p>Darwin, Charles. 1859. <em>The Origin of Species by Means of Natural Selection: Or, the Preservation of Favored Races in the Struggle for Life</em>.</p>
</div>
<div id="ref-Felsenstein:1981vk">
<p>Felsenstein, J. 1981. “Evolutionary trees from DNA sequences: a maximum likelihood approach.” <em>Journal of Molecular Evolution</em> 17 (6): 368–76. <a href="http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;id=7288891&amp;retmode=ref&amp;cmd=prlinks">http://eutils.ncbi.nlm.nih.gov/entrez/eutils/elink.fcgi?dbfrom=pubmed&amp;id=7288891&amp;retmode=ref&amp;cmd=prlinks</a>.</p>
</div>
<div id="ref-haeckel1897">
<p>Haeckel, Ernst. 1897. <em>The Evolution of Man</em>. Vol. 1. D. Appleton.</p>
</div>
<div id="ref-hennig1966">
<p>Hennig, Willi. 1966. <em>Phylogenetic Systematics</em>. University of Illionois Press.</p>
</div>
<div id="ref-Hohna2014">
<p>Hohna, S., T. A. Heath, B. Boussau, M. J. Landis, F. Ronquist, and J. P. Huelsenbeck. 2014. “Probabilistic Graphical Model Representation in Phylogenetics.” Journal Article. <em>Syst Biol</em> 63 (5): 753–71. <a href="https://doi.org/10.1093/sysbio/syu039">https://doi.org/10.1093/sysbio/syu039</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="phylogenies.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["phylogenetic_biology.pdf", "phylogenetic_biology.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
