<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Simulation | Phylogenetic Biology</title>
  <meta name="description" content="An introduction to Phylogenetic Biology." />
  <meta name="generator" content="bookdown 0.20 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Simulation | Phylogenetic Biology" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="An introduction to Phylogenetic Biology." />
  <meta name="github-repo" content="caseywdunn/phylogenetic_biology" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Simulation | Phylogenetic Biology" />
  <meta name="twitter:site" content="@caseywdunn" />
  <meta name="twitter:description" content="An introduction to Phylogenetic Biology." />
  

<meta name="author" content="Casey W. Dunn" />


<meta name="date" content="2020-09-02" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="phylogenies.html"/>
<link rel="next" href="inference.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Phylogenetic Biology</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a><ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#source-code"><i class="fa fa-check"></i><b>0.1</b> Source code</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#software-versions"><i class="fa fa-check"></i><b>0.2</b> Software versions</a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i><b>0.3</b> License</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="intro.html"><a href="intro.html#history-of-the-field"><i class="fa fa-check"></i><b>1.1</b> History of the field</a></li>
<li class="chapter" data-level="1.2" data-path="intro.html"><a href="intro.html#core-concepts"><i class="fa fa-check"></i><b>1.2</b> Core concepts</a></li>
<li class="chapter" data-level="1.3" data-path="intro.html"><a href="intro.html#a-unified-perspective-on-phylogenetic-studies"><i class="fa fa-check"></i><b>1.3</b> A unified perspective on phylogenetic studies</a></li>
<li class="chapter" data-level="1.4" data-path="intro.html"><a href="intro.html#applications"><i class="fa fa-check"></i><b>1.4</b> Applications</a></li>
<li class="chapter" data-level="1.5" data-path="intro.html"><a href="intro.html#additional-resources"><i class="fa fa-check"></i><b>1.5</b> Additional resources</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="phylogenies.html"><a href="phylogenies.html"><i class="fa fa-check"></i><b>2</b> Phylogenies</a><ul>
<li class="chapter" data-level="2.1" data-path="phylogenies.html"><a href="phylogenies.html#phylogenies-are-graphs"><i class="fa fa-check"></i><b>2.1</b> Phylogenies are graphs</a></li>
<li class="chapter" data-level="2.2" data-path="phylogenies.html"><a href="phylogenies.html#drawing-phylogenies"><i class="fa fa-check"></i><b>2.2</b> Drawing phylogenies</a></li>
<li class="chapter" data-level="2.3" data-path="phylogenies.html"><a href="phylogenies.html#some-properties-of-trees"><i class="fa fa-check"></i><b>2.3</b> Some properties of trees</a></li>
<li class="chapter" data-level="2.4" data-path="phylogenies.html"><a href="phylogenies.html#rotating-nodes"><i class="fa fa-check"></i><b>2.4</b> Rotating nodes</a></li>
<li class="chapter" data-level="2.5" data-path="phylogenies.html"><a href="phylogenies.html#the-meaning-of-edge-lengths"><i class="fa fa-check"></i><b>2.5</b> The meaning of edge lengths</a></li>
<li class="chapter" data-level="2.6" data-path="phylogenies.html"><a href="phylogenies.html#naming-nodes-and-groups"><i class="fa fa-check"></i><b>2.6</b> Naming nodes and groups</a></li>
<li class="chapter" data-level="2.7" data-path="phylogenies.html"><a href="phylogenies.html#polytomies"><i class="fa fa-check"></i><b>2.7</b> Polytomies</a></li>
<li class="chapter" data-level="2.8" data-path="phylogenies.html"><a href="phylogenies.html#rooting"><i class="fa fa-check"></i><b>2.8</b> Rooting</a></li>
<li class="chapter" data-level="2.9" data-path="phylogenies.html"><a href="phylogenies.html#the-information-contained-in-phylogenies"><i class="fa fa-check"></i><b>2.9</b> The information contained in phylogenies</a></li>
<li class="chapter" data-level="2.10" data-path="phylogenies.html"><a href="phylogenies.html#representation"><i class="fa fa-check"></i><b>2.10</b> Representation</a></li>
<li class="chapter" data-level="2.11" data-path="phylogenies.html"><a href="phylogenies.html#additional-reading"><i class="fa fa-check"></i><b>2.11</b> Additional reading</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="simulation.html"><a href="simulation.html"><i class="fa fa-check"></i><b>3</b> Simulation</a><ul>
<li class="chapter" data-level="3.1" data-path="simulation.html"><a href="simulation.html#models"><i class="fa fa-check"></i><b>3.1</b> Models</a></li>
<li class="chapter" data-level="3.2" data-path="simulation.html"><a href="simulation.html#a-simple-model-of-evolution"><i class="fa fa-check"></i><b>3.2</b> A simple model of evolution</a><ul>
<li class="chapter" data-level="3.2.1" data-path="simulation.html"><a href="simulation.html#expected-amount-of-change"><i class="fa fa-check"></i><b>3.2.1</b> Expected amount of change</a></li>
<li class="chapter" data-level="3.2.2" data-path="simulation.html"><a href="simulation.html#expected-end-state"><i class="fa fa-check"></i><b>3.2.2</b> Expected end state</a></li>
<li class="chapter" data-level="3.2.3" data-path="simulation.html"><a href="simulation.html#analytical-approach"><i class="fa fa-check"></i><b>3.2.3</b> Analytical approach</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="simulation.html"><a href="simulation.html#generalizing-the-simple-model"><i class="fa fa-check"></i><b>3.3</b> Generalizing the simple model</a></li>
<li class="chapter" data-level="3.4" data-path="simulation.html"><a href="simulation.html#more-complex-models"><i class="fa fa-check"></i><b>3.4</b> More complex models</a></li>
<li class="chapter" data-level="3.5" data-path="simulation.html"><a href="simulation.html#model-structure"><i class="fa fa-check"></i><b>3.5</b> Model structure</a></li>
<li class="chapter" data-level="3.6" data-path="simulation.html"><a href="simulation.html#aditional-resources"><i class="fa fa-check"></i><b>3.6</b> Aditional resources</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="inference.html"><a href="inference.html"><i class="fa fa-check"></i><b>4</b> Inference</a></li>
<li class="chapter" data-level="5" data-path="evaluation.html"><a href="evaluation.html"><i class="fa fa-check"></i><b>5</b> Evaluation</a><ul>
<li class="chapter" data-level="5.1" data-path="evaluation.html"><a href="evaluation.html#model-evaluation"><i class="fa fa-check"></i><b>5.1</b> Model evaluation</a></li>
<li class="chapter" data-level="5.2" data-path="evaluation.html"><a href="evaluation.html#topology-evaluation"><i class="fa fa-check"></i><b>5.2</b> Topology evaluation</a></li>
<li class="chapter" data-level="5.3" data-path="evaluation.html"><a href="evaluation.html#sensitiviy"><i class="fa fa-check"></i><b>5.3</b> Sensitiviy</a></li>
<li class="chapter" data-level="5.4" data-path="evaluation.html"><a href="evaluation.html#confidence"><i class="fa fa-check"></i><b>5.4</b> Confidence</a></li>
<li class="chapter" data-level="5.5" data-path="evaluation.html"><a href="evaluation.html#epistemology"><i class="fa fa-check"></i><b>5.5</b> Epistemology</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="future.html"><a href="future.html"><i class="fa fa-check"></i><b>6</b> Future</a></li>
<li class="chapter" data-level="7" data-path="shape.html"><a href="shape.html"><i class="fa fa-check"></i><b>7</b> Shape</a></li>
<li class="chapter" data-level="8" data-path="character-evolution.html"><a href="character-evolution.html"><i class="fa fa-check"></i><b>8</b> Character evolution</a></li>
<li class="chapter" data-level="9" data-path="intro.html"><a href="intro.html#applications"><i class="fa fa-check"></i><b>9</b> Applications</a><ul>
<li class="chapter" data-level="9.1" data-path="applications.html"><a href="applications.html"><i class="fa fa-check"></i><b>9.1</b> Functional genomic data</a></li>
<li class="chapter" data-level="9.2" data-path="applications.html"><a href="applications.html#gene-trees"><i class="fa fa-check"></i><b>9.2</b> Gene trees</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Phylogenetic Biology</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="simulation" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> Simulation</h1>
<p>Here we will build the machinery to use a model if DNA evolution. Our first
application of the model will be to simulate data.</p>
<div id="models" class="section level2">
<h2><span class="header-section-number">3.1</span> Models</h2>
<p>Models are representations of processes. They are idealized in the sense that they are
deliberately simplified, and they are abstractions in the sense that they leave<br />
things that aren’t thought to be important. As Peter Godfrey-Smith notes <span class="citation">(Godfrey-Smith <a href="#ref-godfrey2016philosophy" role="doc-biblioref">2013</a>,, 21)</span>,
“Abstraction, to some degree, is inevitable; you can’t include everything.
Idealization, in contrast, is a choice.”</p>
<p>For some application, it doesn’t
matter if the model structure reflects the underlying
process as long as it generates useful output <span class="citation">(Breiman and others <a href="#ref-breiman2001statistical" role="doc-biblioref">2001</a>)</span>.
For example, if a large retail chain is trying to predict how much toothpaste it
needs to stock at each store it likely doesn’t care if its models properly consider
purchasing rates and all the other things that impact stock, just as long as the
the model does a reasonable job based on the data at hand of making reasonable
predictions. In science, though, we often care very much about the model because
many of our questions have to do with the mechanisms that underlay the processes
we are modeling. We don’t just want models that give us the right answer, we often
want models that give us the right answer for the right reason.</p>
<p>Statisticians often note that “All models are wrong but some
are useful”, a common aphorism expanded from a quote by George Box <span class="citation">(Box <a href="#ref-box1976science" role="doc-biblioref">1976</a>)</span>.
The goal of a model isn’t to be “right” in the sense of being a perfect explanation of
data. Real data are far too complex for any model to even succeed at this goal, their complexity
would require a tremendous amount of data to feed, and their utility would be undermined by this
complexity. The most useful models strike a trade-off between simplicity and adequacy. They are
as simple as possible, while adequately describing the phenomenon of interest. Adequacy is often
in the eye of the beholder - one scientist will be perfectly happy with a model that makes reasonable
rough approximations of a system, another scientist may be interested in second and third order effects
for which more complexity is needed for adequate explanation.</p>
<p>Cartography is an interesting analogy for the way we will use statistical models. A “perfect” map would basically
be a copy of the whole world, which wouldn’t be that much more useful than the world itself already is for many
of the things you would like to do with a map. So all maps are simplifications (Figure <a href="simulation.html#fig:sim-maps">3.1</a>). That simplification is useful.</p>
<div class="figure"><span id="fig:sim-maps"></span>
<img src="figures/maps.png" alt="Four maps of the Yale campus, varying in complexity and focus. (A) A satellite image of New Haven, including much of Yale campus, from Google Maps. This image has a very large amount of information. (B) A street map of the same region, also from Google Maps. It has less information, but is more useful for some tasks such as navigation. (C) An even more somplified map, focused on showing the Yale Shuttle routes. (D) The New Haven property map of the region around Osborn Memorial Laboratory, showing property lines and plot numbers. Like (C) it is simple, but reflects different decisions about which information to discard or retain. This figure is inspired by the London maps that David Swofford uses in is hown talks to make the same points."  />
<p class="caption">
Figure 3.1: Four maps of the Yale campus, varying in complexity and focus. (A) A satellite image of New Haven, including much of Yale campus, from Google Maps. This image has a very large amount of information. (B) A street map of the same region, also from Google Maps. It has less information, but is more useful for some tasks such as navigation. (C) An even more somplified map, focused on showing the Yale Shuttle routes. (D) The New Haven property map of the region around Osborn Memorial Laboratory, showing property lines and plot numbers. Like (C) it is simple, but reflects different decisions about which information to discard or retain. This figure is inspired by the London maps that David Swofford uses in is hown talks to make the same points.
</p>
</div>
<p>Let’s examine one of the most common models, the linear model:</p>
<p><span class="math display" id="eq:linear">\[\begin{equation} 
  y = mx+b
  \tag{3.1}
\end{equation}\]</span></p>
<p><span class="math inline">\(y\)</span> and <span class="math inline">\(x\)</span> are variables. The model posits a linear relationship between <span class="math inline">\(x\)</span> and <span class="math inline">\(y\)</span>, that is that if you plot their correspondence in a plane
you will get a line. <span class="math inline">\(m\)</span> and <span class="math inline">\(b\)</span> are model parameters. <span class="math inline">\(m\)</span> is the slope of the line, and <span class="math inline">\(b\)</span> is the intercept.</p>
<div class="figure"><span id="fig:sim-linear"></span>
<img src="phylogenetic_biology_files/figure-html/sim-linear-1.png" alt="A linear model with $m$=0.5 and $b$=1." width="2100" />
<p class="caption">
Figure 3.2: A linear model with <span class="math inline">\(m\)</span>=0.5 and <span class="math inline">\(b\)</span>=1.
</p>
</div>
<p>There are a variety of useful things we could do bassed on these relationships between the model, model parameters, and values. Let’s consider <span class="math inline">\(x\)</span> to be variables that tell us something about the past and <span class="math inline">\(y\)</span> to be variables that tell us something about the present or future. Use cases then include:</p>
<ul>
<li><p>If we clamp the model (linear) and model parameters (specific values of <span class="math inline">\(m\)</span> and <span class="math inline">\(b\)</span>) according to prior knowledge, and clamp <span class="math inline">\(y\)</span> according to observed data, we can estimate <span class="math inline">\(x\)</span>. In this case the model is like a time machine that allows us to look into the past.</p></li>
<li><p>If we clamp the model (linear) and variables (<span class="math inline">\(x\)</span> and <span class="math inline">\(y\)</span>) according to prior knowledge, we can estimate model parameters <span class="math inline">\(m\)</span> and <span class="math inline">\(b\)</span>. In this case the model is like an instrument that allows us to peer inside processes based on their inputs and outputs.</p></li>
<li><p>If we clamp the model (linear) and model parameters (specific values of <span class="math inline">\(m\)</span> and <span class="math inline">\(b\)</span>) according to prior knowledge, and clamp <span class="math inline">\(x\)</span> according to observed data, we can estimate <span class="math inline">\(y\)</span>. In this case the model is like an oracle that predicts the future.</p></li>
<li><p>If we clamp the model (linear) and model parameters (specific values of <span class="math inline">\(m\)</span> and <span class="math inline">\(b\)</span>) according to prior knowledge, and clamp <span class="math inline">\(x\)</span> according to values we make up, we can simulate <span class="math inline">\(y\)</span>. In this case the model is like a world builder that tells us what we would expect to see under the specified conditions. This is very helpful for getting a sense of whether our models are working well (Do they generate results that look like what we see?), examining counterfactual situations that don’t exist, or building datasets to test software tools.</p></li>
</ul>
<p>There are some deep connections here. For example, predicting the future is basically just simulating the future based on our model and what we know now.</p>
<p>The models that we will use in phylogenetic biology tend to be more complex than the linear mode, but this general perspective of clamping and estimating different things still holds.</p>
</div>
<div id="a-simple-model-of-evolution" class="section level2">
<h2><span class="header-section-number">3.2</span> A simple model of evolution</h2>
<p>Let’s start with a simple model of DNA evolution. At first we will also consider only a single nucleotide position along a single edge in a phylogeny. The goal is to build an intuitive integrated understanding of the mathematical and statistical relationships among:</p>
<ul>
<li>Model structure</li>
<li>Model parameters</li>
<li>Edge length</li>
<li>State at the start of the edge (the nucleotide at the parent node)</li>
<li>State at the end of the edge (the nucleotide at the child node)</li>
</ul>
<p>Imagine that when the DNA is being replicated, most of the time the appropriate nucleotide is incorporated. Some fraction of the time, at rate <span class="math inline">\(\mu\)</span>, an event occurs where the appropriate nucleotides is replaced with a random nucleotide instead. In our model, the probability of selecting any of the nucleotides during one of these random replacement events is uniform (picking a C is just as probably as picking a G, for example), and the new nucleotide doesn’t depend in any way on what nucleotide was there before. It is as if you had a bag containing equal frequencies of C, G, T, and A nucleotides. As you built the new DNA strand, every so often you would replace the nucleotide you should be adding with one you instead selected by reaching into the bag and picking one at random. We will call the rate at which any one of these nucleotides is placed in a sequence at random <span class="math inline">\(\beta\)</span>, where <span class="math inline">\(\beta=\mu/4\)</span>.</p>
<p>Not all replacement events will result in an apparent change. Sometimes the appropriate nucleotide is selected by chance, even though it was picked at random. If, for example, the appropriate nucleotide was an A, under this model <span class="math inline">\(1/4\)</span> of the time a replacement event occurs an A is selected by chance and there is no apparent change. In such a case, there has not been a substitution (just a replacement in kind). If the A is replaced with any of the other three nucleotides we say there has been a substitution. Because three of the four possible outcomes of an event result in a substitution, the substitution rate is <span class="math inline">\(3\beta\)</span>, which, because <span class="math inline">\(\beta=\mu/4\)</span>, is equivalent to noting that the substitution is <span class="math inline">\((3/4) \mu\)</span>. Because some events result in no apparent change, substitutions are only a subset of events and the substitution rate is <em>lower</em> than the replacement event rate.</p>
<p>It might seem a bit odd to consider replacement events that don’t result in substitutions, but this follows naturally from a central feature we specified for the the model - the new nucleotide doesn’t depend in any way on what nucleotide was there before. If we had a process where replacements always resulted in substitutions, then excluding the a replacement in kind would require knowing which nucleotide should be placed so that we <em>don’t</em> select it.</p>
<div id="expected-amount-of-change" class="section level3">
<h3><span class="header-section-number">3.2.1</span> Expected amount of change</h3>
<p>One of the primary values of a model is that it allows us to think explicitly about how much evolutionary change we expect to see under the specified process. For the simple process described here, there are two things to consider if we want to know the amount of evolutionary change. The first is the substitution rate <span class="math inline">\(\mu\)</span> (which we also know if we know <span class="math inline">\(\beta\)</span>, since <span class="math inline">\(\mu=4\beta\)</span>), and the time over which the evolutionary process acts.</p>
<p>In Figure <a href="simulation.html#fig:sim-jc-mu-sweep">3.3</a> the amount of evolutionary time is held constant, and the rate <span class="math inline">\(\mu\)</span> is changed. When <span class="math inline">\(\mu=0\)</span>, the bottom bar, there are no replacements (black bars) and therefore no substitutions (the whole bar is the same color).</p>
<div class="figure"><span id="fig:sim-jc-mu-sweep"></span>
<img src="phylogenetic_biology_files/figure-html/sim-jc-mu-sweep-1.png" alt="Each horizontal bar is a simulation of evolution of a single nucleotide position through time, $t$, for a specified value of $\mu$. Each sumulation starts out as an A. Black vertical bars correspond to replacement events, which don't all lead to substitutions (a new color). " width="2100" />
<p class="caption">
Figure 3.3: Each horizontal bar is a simulation of evolution of a single nucleotide position through time, <span class="math inline">\(t\)</span>, for a specified value of <span class="math inline">\(\mu\)</span>. Each sumulation starts out as an A. Black vertical bars correspond to replacement events, which don’t all lead to substitutions (a new color).
</p>
</div>
<p>As <span class="math inline">\(\mu=0\)</span> increases (going up on the <span class="math inline">\(y\)</span> axis), the number of replacement events over the same time interval increases (Figure <a href="simulation.html#fig:sim-jc-mu-n">3.4</a>). This reflects the simple linear relationship <span class="math inline">\(n=\mu t\)</span>, where <span class="math inline">\(n\)</span> is the number of expected replacement events.</p>
<div class="figure"><span id="fig:sim-jc-mu-n"></span>
<img src="phylogenetic_biology_files/figure-html/sim-jc-mu-n-1.png" alt="The number of replacement events increases linearly with the replacement rate $\mu$. This plot is from the same simulation as that shown in Figure \@ref(fig:sim-jc-mu-sweep). The line is a linear model fit to the data." width="2100" />
<p class="caption">
Figure 3.4: The number of replacement events increases linearly with the replacement rate <span class="math inline">\(\mu\)</span>. This plot is from the same simulation as that shown in Figure <a href="simulation.html#fig:sim-jc-mu-sweep">3.3</a>. The line is a linear model fit to the data.
</p>
</div>
<p>Because of the linear relationship between the number of replacements and the product <span class="math inline">\(\mu t\)</span>, rate and time are conflated. In many scenarios you can’t estimate them independently. If there are a small number of replacements, for example, you can’t be sure if there is a low rate over a long time interval or a high rate over a short interval. Both would give the same result. Because they are so often confounded in phylogenetic questions, often the rate is essentially fixed at one and the unit of time for edge lengths is given as the number of expected evolutionary change rather than absolute time (years, months, etc). You will often see this length as the scale bar of published phylogenies (Figure <a href="simulation.html#fig:sim-tree-cnid">3.5</a>). The exception is when you have external information, such as dated fossils, that allow you to independently estimate edge lengths and rates.</p>
<div class="figure"><span id="fig:sim-tree-cnid"></span>
<img src="figures/Fig_cnidaria.png" alt="A published phylogeny [@zapata2015] with a scale bar indicating branch length in terms of the expected amount of evolutionary change, rather than absolute time."  />
<p class="caption">
Figure 3.5: A published phylogeny <span class="citation">(Zapata et al. <a href="#ref-zapata2015" role="doc-biblioref">2015</a>)</span> with a scale bar indicating branch length in terms of the expected amount of evolutionary change, rather than absolute time.
</p>
</div>
</div>
<div id="expected-end-state" class="section level3">
<h3><span class="header-section-number">3.2.2</span> Expected end state</h3>
<p>The machinery above shows how a model can clarify the way we think about the expected amount of change. Many times, though, we want to know what the probability of a given end state is given a starting state, a model, and the amount of time elapsed. One way to anchor such a question is to think about the extremes - what do we expect after a very small amount of change (either a short time or a slow rate of change, or both), and what do we expect after a large amount of change?</p>
<p>The situation is most clear after a small amount of change - we expect the end result to be the same as the starting condition. If we start with an A, for example, we expect to end with an A (Figure <a href="simulation.html#fig:sim-saturation">3.6</a>, left side). In this situation, if we know the starting state that information tells us a lot about the end state. Not much else matters.</p>
<p>What should we expect, though, if there has been a large amount of change? Can we know anything at all? It turns out that we can. If there have been many replacements, one after the other, than the initial starting state really doesn’t matter at all because whatever was there will probably have been replaced multiple times. If the starting state doesn’t contain information about the end state, what does? It is the bag that you are picking the nucleotides at random from. Given enough evolutionary time, our simple model will lead the expected frequency of each nucleotide in the evolving sequence to be the same as the frequency in the bag that we randomly draw them from. Since we specified that you have the same chance of grabbing any nucleotide from the bag, eventually the probability of having each of the our nucleotides is the same, 25% (Figure <a href="simulation.html#fig:sim-saturation">3.6</a>, right side). If you started with a sequence that had an A and let it evolve 100 times, after enough evolutionary time had passed to reach equilibrium you would expect to get 25 C’s, 25 G’s, 25 T’s, and 25 A’s.</p>
<div class="figure"><span id="fig:sim-saturation"></span>
<img src="phylogenetic_biology_files/figure-html/sim-saturation-1.png" alt="Stacked bar plots indicating the frequency of each nucleotide after evolution for a specified amount of time. The rate of evolution is $\mu= 0.05 $. There are  1000  replicate simulations for each value of time. At time=0 (no evolution), the end result is always the same as the initial value, which is fixed at A in these simulations. As the length of time increases, the four nucleotides converge on equal frequencies of 25% each." width="2100" />
<p class="caption">
Figure 3.6: Stacked bar plots indicating the frequency of each nucleotide after evolution for a specified amount of time. The rate of evolution is $= 0.05 $. There are 1000 replicate simulations for each value of time. At time=0 (no evolution), the end result is always the same as the initial value, which is fixed at A in these simulations. As the length of time increases, the four nucleotides converge on equal frequencies of 25% each.
</p>
</div>
</div>
<div id="analytical-approach" class="section level3">
<h3><span class="header-section-number">3.2.3</span> Analytical approach</h3>
<p>We are going to use exponential functions in a couple weeks, so let’s have a refresher here.</p>
<p><img src="phylogenetic_biology_files/figure-html/unnamed-chunk-3-1.png" width="2100" /></p>
</div>
</div>
<div id="generalizing-the-simple-model" class="section level2">
<h2><span class="header-section-number">3.3</span> Generalizing the simple model</h2>
<p>rates, equilibrium frequencies</p>
<pre><code>##      [,1]
## [1,]    0
## [2,]    0
## [3,]    0
## [4,]    0</code></pre>
<p>exponentiation</p>
</div>
<div id="more-complex-models" class="section level2">
<h2><span class="header-section-number">3.4</span> More complex models</h2>
</div>
<div id="model-structure" class="section level2">
<h2><span class="header-section-number">3.5</span> Model structure</h2>
</div>
<div id="aditional-resources" class="section level2">
<h2><span class="header-section-number">3.6</span> Aditional resources</h2>
<ul>
<li><p>My own thinking about this material was heavilly influenced by Paul Lewis’s wonderful lectures at the annual Workshop on Molecular Evolution at Woods Hole. Some of his lectures are now available online as part of the excellent <a href="https://www.youtube.com/channel/UCbAzhfySv7nLCrNYqZvBSMg">Phylo Seminar</a>, starting with <a href="https://www.youtube.com/watch?v=1r4z0YJq580&amp;t=2111s" class="uri">https://www.youtube.com/watch?v=1r4z0YJq580&amp;t=2111s</a></p></li>
<li><p>A great introduction to continuous time models by John Huelsenbeck <a href="https://revbayes.github.io/tutorials/dice/" class="uri">https://revbayes.github.io/tutorials/dice/</a></p></li>
</ul>

</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-box1976science">
<p>Box, George EP. 1976. “Science and Statistics.” <em>Journal of the American Statistical Association</em> 71 (356): 791–99. <a href="https://doi.org/10.1080/01621459.1976.10480949">https://doi.org/10.1080/01621459.1976.10480949</a>.</p>
</div>
<div id="ref-breiman2001statistical">
<p>Breiman, Leo, and others. 2001. “Statistical Modeling: The Two Cultures (with Comments and a Rejoinder by the Author).” <em>Statistical Science</em> 16 (3): 199–231. <a href="https://projecteuclid.org/euclid.ss/1009213726">https://projecteuclid.org/euclid.ss/1009213726</a>.</p>
</div>
<div id="ref-godfrey2016philosophy">
<p>Godfrey-Smith, Peter. 2013. <em>Philosophy of Biology</em>. Princeton University Press.</p>
</div>
<div id="ref-zapata2015">
<p>Zapata, Felipe, Freya E Goetz, Stephen A Smith, Mark Howison, Stefan Siebert, Samuel H Church, Steven M Sanders, et al. 2015. “Phylogenomic Analyses Support Traditional Relationships Within Cnidaria.” <em>PloS One</em> 10 (10): e0139068.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="phylogenies.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="inference.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["phylogenetic_biology.pdf", "phylogenetic_biology.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
